### DISCUSSION TITLE
Arguments to click handler created in x-for loop

### DISCUSSION THREAD

**Discussion by markfirmware** [alpinejs/alpine#943] [1. Help] [answered]
https://codepen.io/markfirmware/pen/dypRKYo

Trying to pass data to a button click handler. A button is created for each item in an x-for loop. I want to pass the item from the loop and also a value drawn from x-data.

The problem is, x-data fields and the x-for item are not in scope when the handler is called.

So, how is this done in alpinejs?

FYI there is an additional pre-problem in that codepen cannot even find the handler. In my actual app the handler is found but the arguments cannot be evaluated.

Thanks,
Mark

**Comment by KevinBatdorf** [comment, marked as answer]
That should work. `k` isn't defined in your example either though. See here: https://codepen.io/KevinBatdorf/pen/973ff6bc74c73c11f8b2471c6c74d98e?editors=1011

**Reply by markfirmware** [reply]
Wow. Thanks. This was a reduced case in point. I will go back to my app and investigate more.
